### PR TITLE
Update discount.php to newest version & Added Gift Card support 

### PR DESCRIPTION
**Pull request by Mavlyan**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -32,7 +31,8 @@
   "require-dev": {
     "mygento/coding-standard": "2.*",
     "jakub-onderka/php-parallel-lint": "^0.9.2",
-    "jakub-onderka/php-console-highlighter": "^0.3.2"
+    "jakub-onderka/php-console-highlighter": "^0.3.2",
```
Comment (luckyraul) [inline composer.json]: remove, they are in coding standart

File: .travis.yml [hunk 2]
```diff
@@ -14,8 +14,9 @@ before_script:
 script:
 - php vendor/bin/parallel-lint --exclude vendor .
 - find . -path ./vendor -prune -o -name '*.xml' -print | xargs xmllint --noout
-- php vendor/bin/phpcs .
+- php vendor/bin/phpcs .  --ignore=./Test/*
```
Comment (luckyraul) [inline .travis.yml]: grump

File: .php_cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
+<?php
+$header = <<<EOF
+@author Mygento Team
+@copyright 2016-2018 Mygento (https://www.mygento.ru)
```
Comment (luckyraul) [inline .php_cs]: 2014

File: grumphp.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      - "var_dump("
+      - "exit;"
+    phplint:
+    phpcsfixer2:
+      config: '.php_cs'
+    phpmd:
+      ruleset: ['vendor/mygento/coding-standard/Mygento-Mage2/phpmd.xml']
+    phpcs:
+      warning_severity: null
+    #phpunit:
```
Comment (luckyraul) [inline grumphp.yml]: enable phpunit

File: Test/Unit/bootstrap.php [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+/**
+ * @author Mygento Team
+ * @copyright 2016-2018 Mygento (https://www.mygento.ru)
+ * @package Mygento_Base
+ */
+
+namespace Mygento\Base\Test\Unit;
```
Comment (luckyraul) [inline Test/Unit/bootstrap.php]: do we still need it?

**Review by luckyraul — DISMISSED** [review]

File: .php_cs (hunk 3 above)
Comment (Mavlyan) [reply .php_cs]: Done

File: .travis.yml (hunk 2 above)
Comment (Mavlyan) [reply .travis.yml]: Done

File: Test/Unit/bootstrap.php (hunk 5 above)
Comment (Mavlyan) [reply Test/Unit/bootstrap.php]: Yes we do.

File: composer.json (hunk 1 above)
Comment (Mavlyan) [reply composer.json]: Done

**Review by luckyraul — APPROVED** [review]

**Pull request merged**